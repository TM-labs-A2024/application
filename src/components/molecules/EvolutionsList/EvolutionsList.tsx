import { List } from '@chakra-ui/react'
import Evolution from '@components/atoms/Cards/Evolution'
import { Evolutions } from '@src/types'
import React from 'react'

export default function EvolutionsList({ evolutions }: { evolutions: Evolutions }) {
  return (
    <List spacing={3} className="h-full">
      {evolutions.map((evolution) => (
        <div className="border-b border-black last:border-transparent" key={evolution.href}>
          <Evolution
            href={evolution.href}
            title={evolution.title}
            description={evolution.description}
            comment={evolution.comment}
          />
        </div>
      ))}
    </List>
  )
}
