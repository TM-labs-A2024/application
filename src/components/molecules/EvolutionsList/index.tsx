import EvolutionsList from './EvolutionsList'

export default EvolutionsList
