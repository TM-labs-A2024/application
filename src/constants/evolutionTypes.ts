export const evolutionTypes = [
  {
    id: '1',
    name: 'Evolución'
  },
  {
    id: '2',
    name: 'Hospitalización'
  },
  {
    id: '3',
    name: 'Alta'
  },
  {
    id: '4',
    name: 'Primera consulta'
  }
]
