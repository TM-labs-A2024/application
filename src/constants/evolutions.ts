export const evolution = {
  specialty: {
    description: '',
    id: '',
    name: ''
  },
  author: '',
  title: '',
  description: '',
  'content-format': '',
  created_at: '',
  payload: {
    type: {
      value: '',
      label: ''
    },
    reason: '',
    description: '',
    history: '',
    examination: '',
    summary: '',
    diagnostic: '',
    comments: ''
  }
}
