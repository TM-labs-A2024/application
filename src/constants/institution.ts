export const institution = {
  id: '3fa85f64-5717-4562-b3fc-2c963f66afa1',
  govId: '123456',
  name: 'Clínica Tex Mex',
  credentials: '14578245',
  address: 'Av. Fernandez Peña, 147.',
  email: 'test@test.com',
  phoneNumber: '+582742214441',
  institutionUser: {
    id: '',
    firstname: '',
    lastname: '',
    govId: '',
    birthdate: '',
    email: 'test@test.com',
    phoneNumber: '+582742214441',
    role: '',
    institutionId: ''
  }
}
