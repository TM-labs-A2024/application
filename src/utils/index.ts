import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
