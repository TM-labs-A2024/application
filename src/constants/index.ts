import { doctors } from './doctors'
import { evolutionTypes } from './evolutionTypes'
import { institution } from './institution'
import { institutions } from './institutions'
import {
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
} from './notifications'
import { nurse } from './nurses'
import { patient, sexOptions, statusOptions } from './patient'
import { patients } from './patients'
import { specialties, specialities } from './specialties'
import { specialtyData } from './specialtyData'

export {
  specialties,
  specialities,
  doctors,
  nurse,
  patient,
  sexOptions,
  statusOptions,
  specialtyData,
  evolutionTypes,
  patients,
  institution,
  institutions,
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
}
