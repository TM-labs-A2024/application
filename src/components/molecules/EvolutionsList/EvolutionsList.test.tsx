import { specialityData } from '@src/constants'
import { render, screen } from '@test/utils/index'
import { format } from 'date-fns'
import { es } from 'date-fns/locale/es'

import '@testing-library/jest-dom'
import EvolutionsList from './EvolutionsList'

const props = specialityData.evolutions.map(({ id, date, type, author, reason }) => ({
  href: `/evolucion/1/${id}`,
  title: `${type}: ${format(new Date(date), 'dd, MMMM yyyy', {
    locale: es
  })}`,
  description: `Editado por: ${author}`,
  comment: `PatologÃ­a: ${reason}`
}))

describe('Molecules > EvolutionsList test', () => {
  test('The quantity of children is the expected', () => {
    render(<EvolutionsList evolutions={props} />)

    expect(screen.getAllByTestId('evolution-card')).toHaveLength(9)
  })

  test('Matches the snapshot', () => {
    render(<EvolutionsList evolutions={props} />)

    expect(screen.getByTestId('evolutions-list')).toMatchSnapshot()
  })
})
