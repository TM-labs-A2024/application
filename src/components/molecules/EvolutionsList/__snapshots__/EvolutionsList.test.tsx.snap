// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules > EvolutionsList test Matches the snapshot 1`] = `
<ul
  class="h-full css-1xa84ef"
  data-testid="evolutions-list"
  role="list"
>
  <div
    class="border-b border-black last:border-transparent"
  >
    <a
      class="chakra-link w-full css-ofs0kf"
      data-testid="evolution-card"
      href="/evolucion/1/1"
    >
      <div
        class="mb-1 flex flex-row items-center justify-between"
      >
        <div
          class="flex flex-col items-start gap-1"
        >
          <b
            class="chakra-text css-qqfgvy"
          >
            Evolución 1: 17, septiembre 2024
          </b>
          <div
            class="flex flex-col items-start gap-1"
          >
            <p
              class="chakra-text css-0"
            >
              Editado por: Pedro Pérez
            </p>
            <p
              class="chakra-text css-0"
            >
              Patología: Dolor
            </p>
          </div>
        </div>
        <div />
      </div>
    </a>
  </div>
  <div
    class="border-b border-black last:border-transparent"
  >
    <a
      class="chakra-link w-full css-ofs0kf"
      data-testid="evolution-card"
      href="/evolucion/1/2"
    >
      <div
        class="mb-1 flex flex-row items-center justify-between"
      >
        <div
          class="flex flex-col items-start gap-1"
        >
          <b
            class="chakra-text css-qqfgvy"
          >
            Evolución 2: 17, septiembre 2024
          </b>
          <div
            class="flex flex-col items-start gap-1"
          >
            <p
              class="chakra-text css-0"
            >
              Editado por: Pedro Pérez
            </p>
            <p
              class="chakra-text css-0"
            >
              Patología: Dolor
            </p>
          </div>
        </div>
        <div />
      </div>
    </a>
  </div>
  <div
    class="border-b border-black last:border-transparent"
  >
    <a
      class="chakra-link w-full css-ofs0kf"
      data-testid="evolution-card"
      href="/evolucion/1/3"
    >
      <div
        class="mb-1 flex flex-row items-center justify-between"
      >
        <div
          class="flex flex-col items-start gap-1"
        >
          <b
            class="chakra-text css-qqfgvy"
          >
            Evolución 3: 17, septiembre 2024
          </b>
          <div
            class="flex flex-col items-start gap-1"
          >
            <p
              class="chakra-text css-0"
            >
              Editado por: Pedro Pérez
            </p>
            <p
              class="chakra-text css-0"
            >
              Patología: Dolor
            </p>
          </div>
        </div>
        <div />
      </div>
    </a>
  </div>
  <div
    class="border-b border-black last:border-transparent"
  >
    <a
      class="chakra-link w-full css-ofs0kf"
      data-testid="evolution-card"
      href="/evolucion/1/4"
    >
      <div
        class="mb-1 flex flex-row items-center justify-between"
      >
        <div
          class="flex flex-col items-start gap-1"
        >
          <b
            class="chakra-text css-qqfgvy"
          >
            Evolución 4: 17, septiembre 1997
          </b>
          <div
            class="flex flex-col items-start gap-1"
          >
            <p
              class="chakra-text css-0"
            >
              Editado por: Pedro Pérez
            </p>
            <p
              class="chakra-text css-0"
            >
              Patología: Dolor
            </p>
          </div>
        </div>
        <div />
      </div>
    </a>
  </div>
  <div
    class="border-b border-black last:border-transparent"
  >
    <a
      class="chakra-link w-full css-ofs0kf"
      data-testid="evolution-card"
      href="/evolucion/1/5"
    >
      <div
        class="mb-1 flex flex-row items-center justify-between"
      >
        <div
          class="flex flex-col items-start gap-1"
        >
          <b
            class="chakra-text css-qqfgvy"
          >
            Evolución 5: 17, septiembre 1997
          </b>
          <div
            class="flex flex-col items-start gap-1"
          >
            <p
              class="chakra-text css-0"
            >
              Editado por: Pedro Pérez
            </p>
            <p
              class="chakra-text css-0"
            >
              Patología: Dolor
            </p>
          </div>
        </div>
        <div />
      </div>
    </a>
  </div>
  <div
    class="border-b border-black last:border-transparent"
  >
    <a
      class="chakra-link w-full css-ofs0kf"
      data-testid="evolution-card"
      href="/evolucion/1/6"
    >
      <div
        class="mb-1 flex flex-row items-center justify-between"
      >
        <div
          class="flex flex-col items-start gap-1"
        >
          <b
            class="chakra-text css-qqfgvy"
          >
            Evolución 6: 17, septiembre 1997
          </b>
          <div
            class="flex flex-col items-start gap-1"
          >
            <p
              class="chakra-text css-0"
            >
              Editado por: Pedro Pérez
            </p>
            <p
              class="chakra-text css-0"
            >
              Patología: Dolor
            </p>
          </div>
        </div>
        <div />
      </div>
    </a>
  </div>
  <div
    class="border-b border-black last:border-transparent"
  >
    <a
      class="chakra-link w-full css-ofs0kf"
      data-testid="evolution-card"
      href="/evolucion/1/7"
    >
      <div
        class="mb-1 flex flex-row items-center justify-between"
      >
        <div
          class="flex flex-col items-start gap-1"
        >
          <b
            class="chakra-text css-qqfgvy"
          >
            Evolución 7: 17, septiembre 1997
          </b>
          <div
            class="flex flex-col items-start gap-1"
          >
            <p
              class="chakra-text css-0"
            >
              Editado por: Pedro Pérez
            </p>
            <p
              class="chakra-text css-0"
            >
              Patología: Dolor
            </p>
          </div>
        </div>
        <div />
      </div>
    </a>
  </div>
  <div
    class="border-b border-black last:border-transparent"
  >
    <a
      class="chakra-link w-full css-ofs0kf"
      data-testid="evolution-card"
      href="/evolucion/1/8"
    >
      <div
        class="mb-1 flex flex-row items-center justify-between"
      >
        <div
          class="flex flex-col items-start gap-1"
        >
          <b
            class="chakra-text css-qqfgvy"
          >
            Evolución 8: 17, septiembre 1997
          </b>
          <div
            class="flex flex-col items-start gap-1"
          >
            <p
              class="chakra-text css-0"
            >
              Editado por: Pedro Pérez
            </p>
            <p
              class="chakra-text css-0"
            >
              Patología: Dolor
            </p>
          </div>
        </div>
        <div />
      </div>
    </a>
  </div>
  <div
    class="border-b border-black last:border-transparent"
  >
    <a
      class="chakra-link w-full css-ofs0kf"
      data-testid="evolution-card"
      href="/evolucion/1/9"
    >
      <div
        class="mb-1 flex flex-row items-center justify-between"
      >
        <div
          class="flex flex-col items-start gap-1"
        >
          <b
            class="chakra-text css-qqfgvy"
          >
            Evolución 9: 17, septiembre 1997
          </b>
          <div
            class="flex flex-col items-start gap-1"
          >
            <p
              class="chakra-text css-0"
            >
              Editado por: Pedro Pérez
            </p>
            <p
              class="chakra-text css-0"
            >
              Patología: Dolor
            </p>
          </div>
        </div>
        <div />
      </div>
    </a>
  </div>
</ul>
`
